From e9333fa552337d44ddd2ea669f236fd9c0ec73d0 Mon Sep 17 00:00:00 2001
From: Sergio Aguayo <sergioag@qmailhosting.net>
Date: Mon, 28 Mar 2022 01:58:43 -0500
Subject: [PATCH] Multiple fixes for 2021.10 in USBArmory MkII (Secure Boot)

- Removed fdt_high, as it will cause FDT to be used in-place instead
  of relocated. This makes the system hang just after jumping to the
  kernel.
- Disabled CAAM. Will use software SHA-256 routines. This is because
  the buffers used are not aligned to 64-byte boundaries. NXP also
  states that they don't support using CAAM in U-Boot in their BSP.
- Make CONFIG_SYS_MMC_ENV_DEV not visible in config because it will
  be set depending on the value of CONFIG_SYS_BOOT_DEV_MICROSD or
  CONFIG_SYS_BOOT_DEV_EMMC.
---
 configs/usbarmory-mark-two_defconfig | 1 -
 env/Kconfig                          | 4 ++++
 include/configs/usbarmory-mark-two.h | 1 -
 3 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/configs/usbarmory-mark-two_defconfig b/configs/usbarmory-mark-two_defconfig
index e0628f23b9..eb10e71d5e 100644
--- a/configs/usbarmory-mark-two_defconfig
+++ b/configs/usbarmory-mark-two_defconfig
@@ -52,7 +52,6 @@ CONFIG_OF_SEPARATE=y
 CONFIG_DEFAULT_DEVICE_TREE="imx6ulz-usbarmory"
 
 # Secure/Verified boot
-CONFIG_FSL_CAAM=y
 CONFIG_IMX_HAB=y
 CONFIG_FIT=y
 CONFIG_FIT_SIGNATURE=y
diff --git a/env/Kconfig b/env/Kconfig
index c0dff1fd81..040fde57ac 100644
--- a/env/Kconfig
+++ b/env/Kconfig
@@ -613,6 +613,8 @@ config SYS_RELOC_GD_ENV_ADDR
 	  Relocate the early env_addr pointer so we know it is not inside
 	  the binary. Some systems need this and for the rest, it doesn't hurt.
 
+if !TARGET_USBARMORY_MARK_TWO
+
 config SYS_MMC_ENV_DEV
 	int "mmc device number"
 	depends on ENV_IS_IN_MMC || ENV_IS_IN_FAT || SYS_LS_PPA_FW_IN_MMC || \
@@ -621,6 +623,8 @@ config SYS_MMC_ENV_DEV
 	help
 	  MMC device number on the platform where the environment is stored.
 
+endif
+
 config SYS_MMC_ENV_PART
 	int "mmc partition number"
 	depends on ENV_IS_IN_MMC || ENV_IS_IN_FAT
diff --git a/include/configs/usbarmory-mark-two.h b/include/configs/usbarmory-mark-two.h
index 948d48a570..e7950775bd 100644
--- a/include/configs/usbarmory-mark-two.h
+++ b/include/configs/usbarmory-mark-two.h
@@ -197,7 +197,6 @@
 	"scriptaddr=0x80800000\0"		\
 	"ramdisk_addr_r=0x83000000\0"		\
 	"optee_addr_r=0x83000000\0"		\
-	"fdt_high=0xffffffff\0"			\
 	"initrd_high=0xffffffff\0"		\
 	"bootfile=zImage\0"			\
 	"opteefile=uTee\0"			\
-- 
2.34.1

